### PR TITLE
Add documentation for profiles and ignore options

### PR DESCRIPTION
**Pull request by BackEndTea**
Please don't merge until we are sure on the names of profiles.

### CODE REVIEW

File: src/guide/profiles.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            ]
+        }
+    }
+}
+```
+
+### The Profiles
+
+Currently infection supports the following profiles:
+* `arithmetic`
```
Comment (borNfreee) [inline src/guide/profiles.md]: I would recommend using `@arithmetic` notation (with `@`). What do you think?

File: src/guide/profiles.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* `boolean`
+* `conditional_boundary`
+* `conditional_negotiation`
+* `function_signature`
+* `number`
+* `operator`
+* `return_value`
+* `sort`
+* `zero_iteration`
+* `default` - This is the default profile, which currently contains all mutators, and is used if no mutator or profile is chosen
```
Comment (borNfreee) [inline src/guide/profiles.md]: What about mutators from the future `@nightmare` profile? I would not enable them by default. In this case `..contains all mutators..` will not be true. Should we care about this now or no?

File: src/guide/profiles.md (hunk 1 above)
Comment (BackEndTea) [reply src/guide/profiles.md]: Is allright with me

File: src/guide/profiles.md (hunk 2 above)
Comment (BackEndTea) [reply src/guide/profiles.md]: I decided to let the documentation reflect the current behaviour. If/Once we add the `@nightmare` profile then this needs to be updated.

File: src/guide/profiles.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                "Full\\NameSpaced\\Class"
+            ]
+        }
+    }
+}
+```
+
+### The Profiles
+
+Currently infection supports the following profiles:
```
Comment (borNfreee) [inline src/guide/profiles.md]: Probably, there should be a comma after `Currently`. At least Grammarly plugin reports an issue

File: src/guide/profiles.md (hunk 2 above)
Comment (borNfreee) [reply src/guide/profiles.md]: agreed

**Comment by borNfreee** [comment]
> Please don't merge until we are sure on the names of profiles.

is it ready to be merged @BackEndTea ?

**Comment by BackEndTea** [comment]
Yeah its good for me. If we want to change the names of the profiles we should probably do that before `0.9.0` is released, so people don't have to update their configs if they use it.

**Comment by borNfreee** [comment]
ok, merging for now

**Pull request merged**